app/src/main/java/ch/epfl/skysync/components/RangeTimePicker.kt

```diff
@@ -0,0 +1,147 @@
+package ch.epfl.skysync.components
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableLongStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.times
+
+/** A composable made to select a time duration of a maximum of 24 hours. */
+@Composable
+fun RangeTimePicker(
+    padding: Dp,
+    title: String,
+    showDialog: Boolean,
+    onDismiss: () -> Unit,
+    onConfirm: (Long) -> Unit
+) {
+  if (!showDialog) {
+    return
+  }
+  var timeDuration by remember { mutableLongStateOf(0L) }
+  val listState1 = rememberLazyListState(Int.MAX_VALUE / 2)
```

Comment:
Could use more descriptive cariable name like for example: `hourListState`

Reply:
Done